### PR TITLE
RHEL & 1:1 distro support

### PR DESCRIPTION
**Pull request by istiak101**
WIP

### CODE REVIEW

**Comment by divinity76** [comment]
RHEL+Oracle Linux+Rocky Linux support would be cool (given that RHEL&co has the best support cycle in the business, 10 years support: RHEL7 released 2014 supported until 2024, RHEL8 released 2019 supported until 2029, RHEL9 released 2022 supported until 2032~)

 but it would also be a significant maintenance burden, especially since none of the core developers actually use RHEL* afaik

**Comment by ScIT-Raphael** [comment]
I'm completly open for RHEL support, if someone starts to do the work - as @istiak101 is currently so friendly to do so - some other devs will probaly get focus for hestia which itself will (hopefully) the needed manpower for testing and development.

**Comment by jaapmarcus** [comment]
> RHEL+Oracle Linux+Rocky Linux support would be cool (given that RHEL&co has the best support cycle in the business, 10 years support: RHEL7 released 2014 supported until 2024, RHEL8 released 2019 supported until 2029, RHEL9 released 2022 supported until 2032~)
> 
> but it would also be a significant maintenance burden, especially since none of the core developers actually use RHEL* afaik

There is a huge demand for VestaCP "RHEL" Community  for support for RHEL.  And with the EOL of  Debian 9, (Ubuntu 18.04 and RHEL7 follow soonish) this demand will not stop. 

When we are closer to an "release" I am happy to set up a beta repo and set up a new testing setup for automated tests. 

Yes It might mean some extra work load for I hope the "community" helps with testing / development and providing the support needed

**Comment by istiak101** [comment]
This is only for RHEL and 1:1 fork e.g Rocky Linux, AlmaLinux, Euro Linux. It is not possible for me to support other RPM variants like Fedora, CentOS, Oracle Linux, Opensuse. They have same structure but in various distro and release there are significan't changes in libraries and packages which would need to supported separately. Also I won't be supporting RHEL 7 even though it still have support from RHEL. In RHEL release lifetime it's locked to package versions for stability and in major releases changes are very drastic. So I modified spec to begin RHEL support with version 8.

https://github.com/hestiacp/hestiacp/blob/main/src/rpm/hestia/hestia.spec#L9

I am currently using RHEL 8 and slowly working my way to RHEL 9 as it's getting official repositories from projects like MariaDB, PowerDNS, Dovecot.

File: bin/v-add-firewall-ipset [hunk 1]
```diff
@@ -44,7 +44,11 @@ IPSET_PATH="$HESTIA/data/firewall/ipset"
 
 # Ensure ipset is installed
 if [ -z "$IPSET_BIN" ]; then
-    apt-get --quiet --yes install ipset > /dev/null
+    if [ -f '/etc/redhat-release' ]; then
```
Comment (jaapmarcus) [inline bin/v-add-firewall-ipset:47]: Should be safe to be removed all Hestia installs should 

We should add it to the dependencies instead

File: bin/v-delete-firewall-ipset [hunk 2]
```diff
@@ -42,7 +42,11 @@ check_hestia_demo_mode
 
 # Install ipset package if missing
 if [ -z "$IPSET_BIN" ]; then
-    apt-get --quiet --yes install ipset > /dev/null
+    if [ -f '/etc/redhat-release' ]; then
```
Comment (jaapmarcus) [inline bin/v-delete-firewall-ipset:45]: Same here

File: bin/v-add-firewall-ipset (hunk 1 above)
Comment (istiak101) [reply bin/v-add-firewall-ipset:47]: Do you mean RPM spec or the installer bash script? So this should be removed altogether both for debian and rhel from the above file?

Comment (istiak101) [reply bin/v-add-firewall-ipset:47]: What about conditional installs? If these are set required in package requirements they will not respect these flag.

https://github.com/hestiacp/hestiacp/blob/main/install/hst-install-debian.sh#L251
https://github.com/hestiacp/hestiacp/blob/main/install/hst-install-debian.sh#L900

**Comment by istiak101** [comment]
@jaapmarcus can you disable Drone CI for this specific PR? As I will be doing a lot of pushes this will trigger a lot of builds. I don't want to waste any resources of HestiaCP. I am not sure if Drone can be configured to ignore draft PR's.

**Comment by jaapmarcus** [comment]
Just delete the .drone file in root

**Comment by divinity76** [comment]
if this PR becomes too invasive over a long period of time, significant effort will be spent on just fixing merge conflicts.

May i suggest merging this PR soon, just removing the stuff from README.MD and restore drone.yaml first; and implement RHEL in smaller patches? 

not super important or anything, but wasting time on merge conflicts is a waste

**Comment by jaapmarcus** [comment]
> if this PR becomes too invasive over a long period of time, significant effort will be spent on just fixing merge conflicts.
> 
> May i suggest merging this PR soon, just removing the stuff from README.MD and restore drone.yaml first; and implement RHEL in smaller patches?
> 
> not super important or anything, but wasting time on merge conflicts is a waste

I am fine with it...

As long we don't introduce changes in hst-install.sh / readme.md and so it shouldn't really matter a lot

**Comment by jaapmarcus** [comment]
> if this PR becomes too invasive over a long period of time, significant effort will be spent on just fixing merge conflicts.
> 
> May i suggest merging this PR soon, just removing the stuff from README.MD and restore drone.yaml first; and implement RHEL in smaller patches?
> 
> not super important or anything, but wasting time on merge conflicts is a waste

In this case the  "merge" conflict was caused by .drone.yml that was changed in the main branch / over here so nothing special

**Comment by istiak101** [comment]
> if this PR becomes too invasive over a long period of time, significant effort will be spent on just fixing merge conflicts.
> 
> May i suggest merging this PR soon, just removing the stuff from README.MD and restore drone.yaml first; and implement RHEL in smaller patches?
> 
> not super important or anything, but wasting time on merge conflicts is a waste

I agree with you. And a big PR with too many changes is also time consuming to review. I previously created two small PR but I then thought about making a draft PR to let the community know about the progress while I work on it. I will send small patchsets after this PR. Marking this one ready for review.

**Review by jaapmarcus — APPROVED** [review]

**Comment by ScIT-Raphael** [comment]
Hi @istiak101, sorry to reach out to you that way, but I havent noticed you on our discord. Do you want to join our server and give me a short ping in any of the channels? Would love to get in contact with you. https://discord.gg/nXRUZch

**Pull request merged**